src/main/kotlin/org/rust/ide/formatter/impl/alignment.kt

```diff
@@ -8,6 +8,29 @@ import org.rust.lang.core.psi.RustTokenElementTypes.DOT
 fun RustFmtBlock.getAlignmentStrategy(): RustAlignmentStrategy = when (node.elementType) {
     TUPLE_EXPR, ARG_LIST, FORMAT_MACRO_ARGS, TRY_MACRO_ARGS ->
         RustAlignmentStrategy.wrap()
+            .alignIf { child, parent, ctx ->
+                // Do not align if we have only one argument as this may lead to
+                // some quirks when that argument is tuple expr.
+                // Alignment do not allow "negative indentation" i.e.:
+                //     func((
+                //         happy tuple param
+                //     ))
+                // would be formatted to:
+                //     func((
+                //              happy tuple param
+                //          ))
+                // because due to applied alignment, closing paren has to be
+                // at least in the same column as the opening one.
```

Comment:
Looks reasonable, though maaybe there is something more general?

Like, how

```
foo(1, (
    1,
    2,
    3
))
```

Should be formatted?